### PR TITLE
Make all headers in editors look the same

### PR DESCRIPTION
**Pull request by cndonovan**
See this PR for context: https://github.com/perses/perses/pull/783

Signed-off-by: Christine Donovan <christine.donovan@chronosphere.io>

### CODE REVIEW

File: ui/dashboards/src/components/Variables/VariableEditor.tsx [hunk 1]
```diff
@@ -120,111 +120,118 @@ export function VariableEditor(props: {
   };
 
   return (
-    <Box p={4}>
+    <>
       {currentEditingVariableDefinition && (
-        <>
-          <Typography variant="h3" mb={2}>
-            Edit Variable
-          </Typography>
```
Comment (cndonovan) [inline ui/dashboards/src/components/Variables/VariableEditor.tsx:128]: Moved this into VariableEditForm

File: ui/dashboards/src/components/Variables/VariableEditor.tsx [hunk 2]
```diff
@@ -120,111 +120,118 @@ export function VariableEditor(props: {
   };
 
   return (
-    <Box p={4}>
```
Comment (cndonovan) [inline ui/dashboards/src/components/Variables/VariableEditor.tsx:123]: Moved this padding below the header

File: ui/dashboards/src/components/Variables/VariableEditor.tsx [hunk 3]
```diff
@@ -120,111 +120,118 @@ export function VariableEditor(props: {
...
+                onClick={() => {
+                  props.onCancel();
+                }}
+              >
+                Cancel
               </Button>
-            </Box>
-          </Stack>
+            </Stack>
+          </Box>
```
Comment (cndonovan) [inline ui/dashboards/src/components/Variables/VariableEditor.tsx:166]: This `Box` is the header copied from `PanelDrawer`. I replaced the title and actual buttons.

File: ui/dashboards/src/components/Variables/VariableEditor.tsx [hunk 4]
```diff
@@ -120,111 +120,118 @@ export function VariableEditor(props: {
...
+          <Box padding={2}>
```
Comment (cndonovan) [inline ui/dashboards/src/components/Variables/VariableEditor.tsx:167]: This `Box` is the rest of the form, remaining changes are indentation

File: ui/dashboards/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx [hunk 5]
```diff
@@ -123,173 +123,187 @@ export function VariableEditForm({
...
+          <Button
+            variant="outlined"
+            onClick={() => {
+              onCancel();
+            }}
+          >
+            Cancel
+          </Button>
+        </Stack>
+      </Box>
```
Comment (cndonovan) [inline ui/dashboards/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx:155]: This `Box` is the header copied from `PanelDrawer`. I replaced the title and buttons.

File: ui/dashboards/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx [hunk 6]
```diff
@@ -123,173 +123,187 @@ export function VariableEditForm({
...
+      <Box padding={2}>
```
Comment (cndonovan) [inline ui/dashboards/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx:156]: This `Box` is the rest of the form, remaining changes are indentation

File: ui/dashboards/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx [hunk 7]
```diff
@@ -123,173 +123,187 @@ export function VariableEditForm({
...
-        </Button>
-        <Button
-          variant="outlined"
-          onClick={() => {
-            onCancel();
-          }}
-        >
-          Cancel
-        </Button>
-      </Stack>
```
Comment (cndonovan) [inline ui/dashboards/src/components/Variables/VariableEditorForm/VariableEditorForm.tsx:292]: Moved these buttons up into the header

**Comment by cndonovan** [comment]
cc: @foxbat07

File: ui/dashboards/src/components/Variables/VariableEditor.tsx [hunk 8]
```diff
@@ -120,111 +120,118 @@ export function VariableEditor(props: {
...
-                      </TableCell>
-                    </TableRow>
-                  ))}
-                </TableBody>
-              </Table>
-            </TableContainer>
-            <Box display="flex">
-              <Button onClick={addVariable} variant="contained">
-                Add New Variable
+          <Box
```
Comment (saminzadeh) [inline ui/dashboards/src/components/Variables/VariableEditor.tsx:138]: Once we figure out the right abstraction, I think some common "helper" components would be nice to keep this consistent across features - similar to what @juliepagano did. I think @eunicorn may have started doing some of this for the other too

**Review by saminzadeh — APPROVED** [review]
Awesome - thanks so much for doing this!

File: ui/dashboards/src/components/Variables/VariableEditor.tsx (hunk 8 above)
Comment (eunicorn) [reply ui/dashboards/src/components/Variables/VariableEditor.tsx:138]: yeah we have a `Drawer` component, maybe we should include the header there as well

Comment (cndonovan) [reply ui/dashboards/src/components/Variables/VariableEditor.tsx:138]: Oh I like that - I was thinking about where a shared component for the header could possibly go

@saminzadeh which helper component did Julie make?

**Pull request merged**